mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAModel.scala

```diff
@@ -197,8 +238,86 @@ class LocalLDAModel private[clustering] (
   // TODO:
   // override def topicDistributions(documents: RDD[(Long, Vector)]): RDD[(Long, Vector)] = ???
 
+  /**
+   * Calculate and return per-word likelihood bound, using the `batch` of
+   * documents as evaluation corpus.
+   */
+  // TODO: calcualte logPerplexity over training set online during training, reusing gammad instead
+  // of performing variational inference again in [[bound()]]
+  def logPerplexity(
+      batch: RDD[(Long, Vector)],
+      totalDocs: Long): Double = {
+    val corpusWords = batch
+      .flatMap { case (_, termCounts) => termCounts.toArray }
+      .reduce(_ + _)
+    val subsampleRatio = totalDocs.toDouble / batch.count()
+    val batchVariationalBound = bound(
+      batch,
+      subsampleRatio,
+      docConcentration,
+      topicConcentration,
+      topicsMatrix.toBreeze.toDenseMatrix,
+      gammaShape,
+      k,
+      vocabSize)
+    val perWordBound = batchVariationalBound / (subsampleRatio * corpusWords)
+
+    perWordBound
+  }
+
+
+  /**
+   *  Estimate the variational bound of documents from `batch`:
+   *  E_q[log p(bath)] - E_q[log q(batch)]
+   */
+  private def bound(
```

Comment:
Is this going to be used elsewhere in a later PR?  (If not, it could be part of logPerplexity.)  If so, then can you please make the documentation clearer, even if it's just referencing Equation (16) in the original Online LDA paper?